id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoDraftController.java

```diff
@@ -0,0 +1,171 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.mosip.idrepository.core.constant.AuditEvents;
+import io.mosip.idrepository.core.constant.AuditModules;
+import io.mosip.idrepository.core.constant.IdRepoErrorConstants;
+import io.mosip.idrepository.core.constant.IdType;
+import io.mosip.idrepository.core.dto.IdRequestDTO;
+import io.mosip.idrepository.core.dto.IdResponseDTO;
+import io.mosip.idrepository.core.exception.IdRepoAppException;
+import io.mosip.idrepository.core.helper.AuditHelper;
+import io.mosip.idrepository.core.logger.IdRepoLogger;
+import io.mosip.idrepository.core.security.IdRepoSecurityManager;
+import io.mosip.idrepository.core.spi.IdRepoDraftService;
+import io.mosip.idrepository.core.util.DataValidationUtil;
+import io.mosip.idrepository.identity.validator.IdRequestValidator;
+import io.mosip.kernel.core.logger.spi.Logger;
+import springfox.documentation.annotations.ApiIgnore;
+
+/**
+ * @author Manoj SP
+ *
+ */
+@RestController
+public class IdRepoDraftController {
+
+	private static final String ID_REPO_DRAFT_CONTROLLER = "IdRepoDraftController";
+
+	private final Logger mosipLogger = IdRepoLogger.getLogger(IdRepoDraftController.class);
+
+	@Autowired
+	private IdRequestValidator validator;
+
+	@Autowired
+	private IdRepoDraftService<IdRequestDTO, IdResponseDTO> draftService;
+
+	@Autowired
+	private AuditHelper auditHelper;
+
+	@InitBinder
+	public void initBinder(WebDataBinder binder) {
+		binder.addValidators(validator);
+	}
+
+	@PreAuthorize("hasAnyRole('REGISTRATION_PROCESSOR')")
+	@PostMapping(path = "/createDraft", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<IdResponseDTO> createDraft(@Validated @RequestBody IdRequestDTO request, @ApiIgnore Errors errors)
+			throws IdRepoAppException {
+		try {
+			DataValidationUtil.validate(errors);
+			return new ResponseEntity<>(draftService.createDraft(request), HttpStatus.OK);
+		} catch (IdRepoAppException e) {
+			auditHelper.auditError(AuditModules.ID_REPO_CORE_SERVICE, AuditEvents.CREATE_DRAFT_REQUEST_RESPONSE,
+					request.getRequest().getRegistrationId(), IdType.RID, e);
+			mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_DRAFT_CONTROLLER, "createDraft", e.getMessage());
+			throw new IdRepoAppException(e.getErrorCode(), e.getErrorText(), e);
+		} finally {
+			auditHelper.audit(AuditModules.ID_REPO_CORE_SERVICE, AuditEvents.CREATE_DRAFT_REQUEST_RESPONSE,
+					request.getRequest().getRegistrationId(), IdType.RID, "Create draft requested");
+		}
+	}
+
+	@PreAuthorize("hasAnyRole('REGISTRATION_PROCESSOR')")
+	@PatchMapping(path = "/updateDraft", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
can we not have all the endpoints as draft/update draft/create draft/publish/:id

Reply:
@gsasikumar I have updated the uri path as suggested